File: 977.squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        size = len(nums)
        for i, num in enumerate(nums):
            nums[i] = num * num;

        squares = []
        l = 0
        r = size - 1

        while l <= r:
            if nums[l] > nums[r]:
                squares.append(nums[l])
                l += 1
            else:
                squares.append(nums[r])
                r -= 1

        squares.reverse()
        return squares

File: 977.squares-of-a-sorted-array.java
class Solution {
    int abs(int num) {
        if (num < 0)
            return num * -1;
        return num;
    }

    public int[] sortedSquares(int[] nums) {
        int i, min, l, r;
        min = 0;
        for (i = 1; i < nums.length; ++i) {
            if (abs(nums[min]) > abs(nums[i]))
                min = i;
        }

        int result[] = new int[nums.length];
        i = 0;
        result[i] = nums[min] * nums[min];
        i++;

        l = min - 1;
        r = min + 1;
        while(l >= 0 || r < nums.length) {
            if (l < 0 || (r < nums.length && (nums[r] * nums[r] < nums[l] * nums[l]))) {
                result[i] = nums[r] * nums[r];
                r++;
            }
            else {
                result[i] = nums[l] * nums[l];
                l--;
            }
            i++;
        }
        return result;
    }
}
